### PR TITLE
Create 02-move-metadata-data-auth-backends-in-arsenal

### PR DESCRIPTION
**Pull request by vrancurel**
This is a draft for a proposal in Zenko.

### CODE REVIEW

File: 02-move-metadata-data-auth-backends-in-arsenal.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+We can assume it will be much simpler for people to contribute new backends in Arsenal, because:
+
+* We will provide a much cleaner documentation for writing a metadata, data or auth backends
+* Contributors could test their backends unitarily in Arsenal (e.g. without having to also test the S3 protocol layer front-end).
+
+### Proposed Changes
+
+#### Architecture Diagram
+#### Components
```
Comment (smaffulli) [inline 02-move-metadata-data-auth-backends-in-arsenal.md]: are you planning on adding these later?

File: 02-move-metadata-data-auth-backends-in-arsenal.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# ZIP#2 (Zenko Improvement Proposal #2) Move Metadata/Data/Auth Backends To Arsenal
+
+## Overview
+
+This document is a proposal for an architecture change on the metadata, data and auth backends of the Cloudserver component.
```
Comment (smaffulli) [inline 02-move-metadata-data-auth-backends-in-arsenal.md]: line wrapping is off. Does it make sense to break at 70 char?

File: 02-move-metadata-data-auth-backends-in-arsenal.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+Currently Cloudserver requires 3 types of backends:
+* Metadata backends: Backends required to store the buckets metadata (e.g. MongoDB).
+* Data backends: Backends required to store the objects data (e.g. local files or public clouds).
+* Auth backends: Backends required to store the credentials such as access and secret keys (e.g. in local files).
+
+Note data backends are separated in 2 sub-classes:
+* By key backends: Very simple backends which does not preserve namespace (e.g. key/value stores).
+* By path backends: More complex backends which preserve the namespace (the path of objects), e.g. public clouds, NAS filers, etc.
+
+Currently metadata backends have already been moved to Arsenal, the common library between all Zenko nodejs based
```
Comment (smaffulli) [inline 02-move-metadata-data-auth-backends-in-arsenal.md]: Maybe add a link to Arsenal git repository?

**Comment by vrancurel** [comment]
@rahulreddy please mention other potential problems

**Comment by vrancurel** [comment]
We would also move auth, and maybe kms in Arsenal.

- cloudserver/lib/auth should land in arsenal/lib/storage/auth
- while arsenal/lib/auth (generic auth routines) shall remain there

**Pull request closed without merging**